docs/source/developer/release.rst

```diff
@@ -0,0 +1,51 @@
+Numba Release Checklist Templates
+==================================
+
+The goal of the Numba release process -- from a birds eye perspective -- is to
+publish source and binary artifacts that correspond to the given version
+number. This usually involves a sequence of individual tasks that must be
+performed in the correct order and with diligence. Numba and llvmlite are
+commonly released in lockstep since there is usually a one-to-one mapping
+between a Numba version and a corresponding llvmlite version.
+
+This section contains various notes and templates that can be used to create a
+Numba release checklist on the Numba Github issue tracker. This is the aid for
+the maintainers during the release process and helps to ensure that all tasks
+are completed in the correct order and that no tasks are accidentally omitted.
+
+Historically, the release checklists were copied from one release to the next.
+While this worked OKish for many years, often times tasks were lost during
+repeated copy and paste. As a result this document was created to keep track of
+all potential tasks. All templates within this document are written in Github
+Markdown and are available as issue templates on Github. This means,
```

Comment:
Why are they called `.yml` as files in the doc source tree if they are markdown `.md`?

Reply:
Because the format of the templates is such that you write markdown for the template content and then wrap that in yaml to add metadata such as title etc...

Reply:
So, this piece of documentation threw me off:

```
Issue templates are stored on the repository's default branch, in a hidden .github/ISSUE_TEMPLATE directory. If you create a template in another branch, it will not be available for collaborators to use. Issue template filenames are not case sensitive, and need a .md extension. To be included in the community profile checklist, issue templates must be located in the .github/ISSUE_TEMPLATE folder and contain valid name: and about: YAML front matter.
```

From: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates 

Will rename the files to `.md`.

Reply:
done